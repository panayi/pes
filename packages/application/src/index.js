/* @flow */
import api from './api';

export { api };
export * from './triggers/adImages';
export * from './triggers/ads';
export * from './triggers/betaInvites';
export * from './triggers/messages';
export * from './triggers/pendingReviewAds';
export * from './triggers/users';
export * from './triggers/storage';
export * from './triggers/authentication';
export * from './triggers/support';
