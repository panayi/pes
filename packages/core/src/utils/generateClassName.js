const generateClassName = () => btoa(Math.random());

export default generateClassName;
