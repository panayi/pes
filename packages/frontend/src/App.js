import React from 'react';
import { ConnectedRouter } from 'react-router-redux';
import { create } from 'jss';
import JssProvider from 'react-jss/lib/JssProvider';
import jssExpand from 'jss-expand';
import {
  MuiThemeProvider,
  createGenerateClassName,
  jssPreset,
} from 'material-ui/styles';
import 'normalize.css/normalize.css';
import theme from 'config/theme';
import { constants as searchConstants } from 'store/search';
import SearchProvider from 'components/SearchProvider/SearchProvider';
import Pages from './pages';

const jss = create({ plugins: [...jssPreset().plugins, jssExpand()] });
const generateClassName = createGenerateClassName();

const App = ({ store }) => (
  <ConnectedRouter history={store.history}>
    <SearchProvider id={searchConstants.HOME_SEARCH_ID}>
      <JssProvider jss={jss} generateClassName={generateClassName}>
        <MuiThemeProvider theme={theme}>
          <Pages />
        </MuiThemeProvider>
      </JssProvider>
    </SearchProvider>
  </ConnectedRouter>
);

export default App;
