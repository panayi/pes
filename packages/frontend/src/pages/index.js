import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import ModalProvider from './components/ModalProvider/ModalProvider';
import Home from './home';
import Profile from './profile';
import Ad from './ad';
import Messages from './messages';

const Pages = () => (
  <React.Fragment>
    <Helmet>
      <title>Buy and sell stuff in Cyprus - Pesposa</title>
      <link rel="canonical" href={window.location.href} />
      <meta
        name="description"
        content="Sell your stuff quickly and connect with thousands of buyers. Find cars, houses, electronics and much more, near your location."
      />
    </Helmet>
    <Switch>
      <Route path="/profile" component={Profile} />
      <Route path="/user/:userId" component={Profile} />
      <Route path="/i/:adId" component={Ad} />
      <Route path="/il/:adId" render={props => <Ad {...props} legacy />} />
      <Route path="/messages" component={Messages} />
      <Route path="/" component={Home} />
    </Switch>
    <ModalProvider />
  </React.Fragment>
);

export default Pages;
