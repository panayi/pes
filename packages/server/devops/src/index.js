import * as R from 'ramda';
import program from 'commander';
import readlineSync from 'readline-sync';
import logger from 'winston-color';
import chalk from 'chalk';
import validateDeployment from './utils/validateDeployment';
import * as commands from './commands';

const COMMAND_ARGS_LENGTH = 2;

program
  .name('yarn devops')
  .option(
    '-p, --production',
    'Run on production deployment (pesposa-production)',
  )
  .command('help', '')
  .description('Usage information')
  .action(() => program.help());

R.forEach(command => command(program), R.values(commands));
program.parse(process.argv);

const requestedProductionDeployment = program.production;

if (!validateDeployment(requestedProductionDeployment)) {
  process.exit();
}

if (requestedProductionDeployment) {
  logger.warn('WARNING! This command will run on production project.');
  const projectName = readlineSync.question(
    'Enter the project name to continue: ',
  );

  if (projectName !== 'pesposa-production') {
    logger.error('Production project name is not correct. Exiting...');
    process.exit(1);
  }
}

logger.info(
  `Running on ${
    requestedProductionDeployment
      ? chalk.yellowBright.inverse('production')
      : chalk.blue('development')
  } deployment`,
);

const commandNames = R.keys(commands);
const command = R.compose(
  R.head,
  R.defaultTo([]),
)(process.argv.slice(COMMAND_ARGS_LENGTH));

if (!command || !R.contains(command, commandNames)) {
  logger.error(`Command "${command || ''}" not found.`);
  program.outputHelp();
  process.exit();
}
