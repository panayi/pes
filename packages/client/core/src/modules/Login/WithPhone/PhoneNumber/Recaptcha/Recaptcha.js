import React, { Component } from 'react';
import * as R from 'ramda';
import { Helmet } from 'react-helmet';
import generateClassName from '@pesposa/core/src/utils/generateClassName';

// type Props = {
//   firebase: Object,
// };

export default class Recaptcha extends Component {
  RECAPTCHA_ID = generateClassName();

  verifier = null;

  componentDidMount() {
    this.createRecaptcha();
  }

  shouldComponentUpdate() {
    return false;
  }

  reset = () =>
    this.verifier.render().then(widgetId => {
      const reset = R.path(['grecaptcha', 'reset'], window);
      if (reset) {
        reset(widgetId);
      }
    });

  createRecaptcha() {
    const { firebase } = this.props;
    const { RecaptchaVerifier } = firebase.auth;
    this.verifier = new RecaptchaVerifier(this.RECAPTCHA_ID, {
      size: 'invisible',
    });
  }

  render() {
    return (
      <div>
        <Helmet>
          <script src="https://www.google.com/recaptcha/api.js" async defer />
        </Helmet>
        <div id={this.RECAPTCHA_ID} />
      </div>
    );
  }
}
