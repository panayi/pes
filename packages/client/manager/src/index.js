import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import registerServiceWorker from './lib/registerServiceWorker';

ReactDOM.render(<App />, document.getElementById('root'));
registerServiceWorker();
