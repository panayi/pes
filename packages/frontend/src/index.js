import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import configureStore from 'store/configureStore';
import registerServiceWorker from 'lib/registerServiceWorker';
import App from './App';

const store = configureStore(window.__INITIAL_STATE__); // eslint-disable-line no-underscore-dangle

ReactDOM.render(
  <Provider store={store}>
    <App store={store} />
  </Provider>,
  document.getElementById('root'),
);

if (module.hot) {
  module.hot.accept('./App', () => {
    ReactDOM.render(
      <Provider store={store}>
        <App store={store} />
      </Provider>,
      document.getElementById('root'),
    );
  });
}

registerServiceWorker();
