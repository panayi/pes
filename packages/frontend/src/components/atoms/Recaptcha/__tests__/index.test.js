import React from 'react';
import * as generate from 'pesposa-utils/src/generateClassName';
import Recaptcha from '../index';

describe('[Component] Recaptcha', () => {
  const defaultProps = {
    firebase: {
      auth: {
        RecaptchaVerifier: Object,
      },
    },
  };

  it('should render correctly', () => {
    generate.default = jest.fn();
    generate.default.mockReturnValueOnce('myRecaptchaId');
    const wrapper = shallow(<Recaptcha {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
    generate.default.mockClear();
  });

  it('should have a unique RECAPTCHA_ID', () => {
    generate.default = jest.fn();
    generate.default.mockReturnValueOnce('first').mockReturnValueOnce('second');

    const wrapper = shallow(<Recaptcha {...defaultProps} />);
    const wrapper2 = shallow(<Recaptcha {...defaultProps} />);

    expect(wrapper.instance().RECAPTCHA_ID).toBe('first');
    expect(wrapper2.instance().RECAPTCHA_ID).toBe('second');
  });

  describe('reset', () => {
    class RecaptchaVerifier {
      render = () => Promise.resolve('myWidgetId');
    }
    const props = {
      firebase: {
        auth: {
          RecaptchaVerifier,
        },
      },
    };

    it('should reset grecaptcha', () => {
      const mockReset = jest.fn();
      window.grecaptcha = {
        reset: mockReset,
      };

      const wrapper = shallow(<Recaptcha {...props} />);

      return wrapper
        .instance()
        .reset()
        .then(() => {
          expect(mockReset.mock.calls).toEqual([['myWidgetId']]);
        });
    });
  });
});
