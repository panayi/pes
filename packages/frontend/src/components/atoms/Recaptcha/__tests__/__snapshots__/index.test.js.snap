// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Recaptcha should render correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      async={true}
      defer={true}
      src="https://www.google.com/recaptcha/api.js"
    />
  </HelmetWrapper>
  <div
    id="myRecaptchaId"
  />
</div>
`;
